rllib/evaluation/rollout_worker.py

```diff
@@ -1270,22 +1279,43 @@ def add_policy(
         if policy_state:
             new_policy.set_state(policy_state)
 
-        filter_shape = tree.map_structure(
-            lambda s: (
-                None
-                if isinstance(s, (Discrete, MultiDiscrete))  # noqa
-                else np.array(s.shape)
-            ),
-            new_policy.observation_space_struct,
-        )
+        # Enabling connectors is not per policy, so we can use the general config here
+        if not connectors_enabled:
+            filter_shape = tree.map_structure(
+                lambda s: (
+                    None
+                    if isinstance(s, (Discrete, MultiDiscrete))  # noqa
+                    else np.array(s.shape)
+                ),
+                new_policy.observation_space_struct,
+            )
+
+            self.filters[policy_id] = get_filter(
+                (config or {}).get("observation_filter", "NoFilter"), filter_shape
+            )
 
-        self.filters[policy_id] = get_filter(self.observation_filter, filter_shape)
+        if connectors_enabled and policy_id in self.policy_map:
+            create_connectors_for_policy(self.policy_map[policy_id], self.policy_config)
+
+            # As long as the historic filter synchronization mechanism is in
+            # place, we need to put filters into self.filters so that they get
+            # synchronized
+            filter_connectors = self.policy_map[policy_id].agent_connectors[
+                SyncedFilterAgentConnector
```

Comment:
What about the concurrent version of the filter?

Reply:
I don't think that has practical relevance but is just used for testing (ConcurrentMeanStdFilter is only used in test_rollout_worker and I can't think of a reason why it should be used elsewhere).
So my idea was rewrite that test to use the SyncedFilterAgentConnector instead of the ConcurrentMeanStdFilter after we make the switch and eliminate ConcurrentMeanStdFilter altogether. 

-> Better get rid of ConcurrentMeanStdFilter because it's just used only for testing and instead test RolloutWorker as a ray actor. Does that make sense?

Reply:
ok, can you add a Note or TODO here, saying that nobody should use ConcurrentMeanStdFilter at this point, since we will remove it?

Reply:
Created a deprecation warning